tests/pytests/unit/modules/test_schedule.py

```diff
@@ -44,19 +44,20 @@ def test_purge(sock_dir, job1):
     Test if it purge all the jobs currently scheduled on the minion.
     """
     _schedule_data = {"job1": job1}
-    with patch.dict(schedule.__opts__, {"schedule": {}, "sock_dir": sock_dir}):
-        mock = MagicMock(return_value=True)
-        with patch.dict(schedule.__salt__, {"event.fire": mock}):
-            _ret_value = {"complete": True, "schedule": {}}
-            with patch.object(SaltEvent, "get_event", return_value=_ret_value):
-                with patch.object(
-                    schedule, "list_", MagicMock(return_value=_schedule_data)
-                ):
-                    assert schedule.purge() == {
-                        "comment": ["Deleted job: job1 from schedule."],
-                        "changes": {"job1": "removed"},
-                        "result": True,
-                    }
+    with patch("os.makedirs", MagicMock(return_value=True)):
+        with patch.dict(schedule.__opts__, {"schedule": {}, "sock_dir": sock_dir}):
+            mock = MagicMock(return_value=True)
+            with patch.dict(schedule.__salt__, {"event.fire": mock}):
+                _ret_value = {"complete": True, "schedule": {}}
+                with patch.object(SaltEvent, "get_event", return_value=_ret_value):
+                    with patch.object(
+                        schedule, "list_", MagicMock(return_value=_schedule_data)
+                    ):
```

Comment:
**suggestion (non-blocking)** to avoid super nested patches, creating the patches first and then using them in a single `with` is :100: 
example:
```
patch_salt = patch.dict(schedule.__salt__, {"event.fire": ...})
patch_salt_event = patch.object(SaltEvent, ...)
patch_schedule_list = patch.object(schedule, 'list_', ...)
with patch_salt, patch_salt_event, patch_schedule_list:
    assert schedule.purge() == {...}
```

Not a big deal, I write mine that way because I feel it's cleaner to me.